HPC/Gromacs.md

```diff
@@ -0,0 +1,95 @@
+* Hpc7g/c7gn (-DGMX_SIMD=ARM_SVE)
+* c6gn (-DGMX_SIMD=ARM_NEON_ASIMD)
+We noticed that on the same Hpc7g instance, which supports both SVE and NEON, choose ARM_SVE as SIMD option gives 10% better performance than ARM_NEON_ASIMD. 
+
+## Build Gromacs
+* follow the instructions to install [ArmPL](README.md#libraries) and build Gromacs with ArmPL fft library
+```
+export LDFLAGS="-lgfortran -lamath -lm -lastring"
+cmake .. -DGMX_BUILD_OWN_FFTW=OFF \
+      -DREGRESSIONTEST_DOWNLOAD=ON \
+      -DCMAKE_C_FLAGS="-mcpu=neoverse-512tvb --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_CXX_FLAGS="-mcpu=neoverse-512tvb --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_C_COMPILER=$(which mpicc) \
+      -DCMAKE_CXX_COMPILER=$(which mpicxx) \
+      -DGMX_OMP=ON \
+      -DGMX_MPI=ON \
+      -DGMX_SIMD=ARM_SVE \
+      -DGMX_BUILD_MDRUN_ONLY=OFF \
+      -DGMX_DOUBLE=OFF \
+      -DCMAKE_INSTALL_PREFIX=${CURDIR} \
+      -DBUILD_SHARED_LIBS=OFF \
+      -DGMX_FFT_LIBRARY=fftw3 \
+      -DFFTWF_LIBRARY=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      -DFFTWF_INCLUDE_DIR=${ARMPL_INCLUDES} \
+      \
+      -DGMX_BLAS_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      -DGMX_LAPACK_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      \
+      -DGMX_GPLUSPLUS_PATH=$(which g++) \
+      -DGMXAPI=OFF \
+      -DGMX_GPU=OFF
+
+```
+* configure Gromacs with FFTW
+```
+cmake .. -DGMX_BUILD_OWN_FFTW=OFF \
+      -DREGRESSIONTEST_DOWNLOAD=ON \
+      -DCMAKE_C_FLAGS="-mcpu=native --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_CXX_FLAGS="-mcpu=native --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_C_COMPILER=$(which mpicc) \
+      -DCMAKE_CXX_COMPILER=$(which mpicxx) \
+      -DGMX_OMP=ON \
+      -DGMX_MPI=ON \
+      -DGMX_SIMD=ARM_SVE \
+      -DGMX_BUILD_MDRUN_ONLY=OFF \
+      -DGMX_DOUBLE=OFF \
+      -DCMAKE_INSTALL_PREFIX=${CURDIR} \
+      -DBUILD_SHARED_LIBS=OFF \
+      -DGMX_FFT_LIBRARY=fftw3 \
+      -DFFTWF_LIBRARY=/shared/fftw-3.3.9/lib/libfftw3f.a \
+      -DFFTWF_INCLUDE_DIR=/shared/fftw-3.3.9/include \
+      \
+      -DGMX_BLAS_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      -DGMX_LAPACK_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      \
+      -DGMX_GPLUSPLUS_PATH=$(which g++) \
+      -DGMXAPI=OFF \
+      -DGMX_GPU=OFF
+```
+
+## Run Gromacs with ArmPL
+slurm script used for running Gromacs job on ParallelCluster.
+```
+#!/bin/bash
+#SBATCH --wait-all-nodes=1
+#SBATCH --ntasks-per-node=64
+#SBATCH --ntasks=64
+#SBATCH --cpus-per-task=1
+#SBATCH --ntasks-per-core=1
+#SBATCH --export=ALL
+#SBATCH --partition=compute
+#SBATCH --exclusive
+
+export PATH=/shared/gromacs-2022.4/bin:$PATH
+export LD_LIBRARY_PATH=/shared/gromacs-2022.4/lib:$LD_LIBRARY_PATH
+export PATH=/shared/gcc-12.2.0/bin:$PATH
```

Comment:
The the time when I did the comparison. Arm is not offerring one for gcc-10.2.0. I use 12.2.0 because I already have it installed on my cluster and it is better to use the latest.  for Gromacs, compiler version doesn't make a big difference to performance.

Reply:
Arm don't provide ArmPL for all compiler versions.

Reply:
Please make this documentation as easy and low-frustration as possible, it should not be tied to something you have lying around, it has to work for someone just starting out.  A script they can run would be even better to pull in the recommended libraries and compilers.